Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -0,0 +1,234 @@
+    if let parent = parent {
+      return .member(name: name, baseType: parent)
+    } else {
+      return .simple(name: name)
+    }
+  }
+
+  private var caseDescription: String {
+    switch self {
+    case .array:
+      return Self.arrayDescription
+    case .composition:
+      return Self.compositionDescription
+    case .dictionary:
+      return Self.dictionaryDescription
+    case .implicitlyUnwrapped:
+      return Self.implicitlyUnwrappedDescription
+    case .member:
+      return Self.memberDescription
+    case .optional:
+      return Self.optionalDescription
+    case .simple:
+      return Self.simpleDescription
+    case .tuple:
+      return Self.tupleDescription
+    case .unknown:
+      return Self.unknownDescription
+    }
+  }
+
+  private static let simpleDescription = "simple"
+  private static let memberDescription = "member"
+  private static let compositionDescription = "composition"
+  private static let optionalDescription = "optional"
+  private static let implicitlyUnwrappedDescription = "implicitlyUnwrapped"
+  private static let arrayDescription = "array"
+  private static let dictionaryDescription = "dictionary"
+  private static let tupleDescription = "tuple"
+  private static let unknownDescription = "unknown"
+}
+
+extension TypeSyntax {
+
+  /// Returns the type description for the receiver.
+  /// - Warning: Do not call on a type syntax node of type `ClassRestrictionTypeSyntax`,
+  ///            `SomeTypeSyntax`, `MetatypeTypeSyntax`, `FunctionTypeSyntax`,
+  ///            `AttributedTypeSyntax`, or `UnknownTypeSyntax`
+  var typeDescription: TypeDescription {
+    if let typeIdentifier = self.as(SimpleTypeIdentifierSyntax.self) {
+      return .simple(name: typeIdentifier.name.text)
+
+    } else if let typeIdentifier = self.as(MemberTypeIdentifierSyntax.self) {
+      return .member(
+        name: typeIdentifier.name.text,
+        baseType: typeIdentifier.baseType.typeDescription)
+
+    } else if let typeIdentifiers = self.as(CompositionTypeSyntax.self) {
+      return .composition(typeIdentifiers.elements.map { $0.type.typeDescription })
+
+    } else if let typeIdentifier = self.as(OptionalTypeSyntax.self) {
+      return .optional(typeIdentifier.wrappedType.typeDescription)
+
+    } else if let typeIdentifier = self.as(ImplicitlyUnwrappedOptionalTypeSyntax.self) {
+      return .implicitlyUnwrapped(typeIdentifier.wrappedType.typeDescription)
+
+    } else if let typeIdentifier = self.as(ArrayTypeSyntax.self) {
+      return .array(typeIdentifier.elementType.typeDescription)
+
+    } else if let typeIdentifier = self.as(DictionaryTypeSyntax.self) {
+      return .dictionary(
+        key: typeIdentifier.keyType.typeDescription,
+        value: typeIdentifier.valueType.typeDescription)
+
+    } else if let typeIdentifiers = self.as(TupleTypeSyntax.self) {
+      return .tuple(typeIdentifiers.elements.map { $0.type.typeDescription })
+
+    } else {
```

Comment:
I currently do not have coverage on this `else` statement. See [this comment](https://github.com/fdiaz/SwiftInspector/pull/69/files#r569045995) for more info.

Reply:
I see some value in verifying the assertion so we know that we have that guardrail in place if the language expands over time and we no longer cover all cases.

Reply:
Yeah the tough thing is I'm running low on ways to create this failing test 😅

Reply:
I will defer to @fdiaz on whether meeting the code coverage requirement is a dealbreaker.

Reply:
I think it's fine, it's an edge case so I'm ok merging through the failing CI check (if that's possible, I'm not sure). If not, we can lower the allowed number per PR to 0.3%